onnxruntime/core/providers/cuda/generator/random.h

```diff
@@ -26,6 +26,7 @@ RANDOM_COMPUTE_IMPL(RandomUniform)
 
 #undef RANDOM_COMPUTE_IMPL
 
+template <typename RandomType>
```

Comment:
Let's try avoid unnecessary templating. This class simply provides functionality that is common and does not depend on any types.